[2025-04-22T17:14:25.432+0000] {taskinstance.py:1957} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: gen_lxw_fts.calculate_fts scheduled__2025-03-08T23:00:00+00:00 [queued]>
[2025-04-22T17:14:25.459+0000] {taskinstance.py:1957} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: gen_lxw_fts.calculate_fts scheduled__2025-03-08T23:00:00+00:00 [queued]>
[2025-04-22T17:14:25.459+0000] {taskinstance.py:2171} INFO - Starting attempt 7 of 8
[2025-04-22T17:14:25.969+0000] {taskinstance.py:2192} INFO - Executing <Task(BashOperator): calculate_fts> on 2025-03-08 23:00:00+00:00
[2025-04-22T17:14:25.983+0000] {standard_task_runner.py:60} INFO - Started process 3609 to run task
[2025-04-22T17:14:25.991+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'gen_lxw_fts', 'calculate_fts', 'scheduled__2025-03-08T23:00:00+00:00', '--job-id', '144', '--raw', '--subdir', 'DAGS_FOLDER/gen_lxw_fts.py', '--cfg-path', '/tmp/tmp_t0dqmnq']
[2025-04-22T17:14:25.993+0000] {standard_task_runner.py:88} INFO - Job 144: Subtask calculate_fts
[2025-04-22T17:14:26.452+0000] {task_command.py:423} INFO - Running <TaskInstance: gen_lxw_fts.calculate_fts scheduled__2025-03-08T23:00:00+00:00 [running]> on host 45b38884c639
[2025-04-22T17:14:27.068+0000] {warnings.py:109} WARNING - /home/***/.local/lib/python3.8/site-packages/***/utils/context.py:207: AirflowContextDeprecationWarning: Accessing 'execution_date' from the template is deprecated and will be removed in a future version. Please use 'data_interval_start' or 'logical_date' instead.
  warnings.warn(_create_deprecation_warning(key, self._deprecation_replacements[key]))

[2025-04-22T17:14:27.229+0000] {taskinstance.py:2481} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='longnv' AIRFLOW_CTX_DAG_ID='gen_lxw_fts' AIRFLOW_CTX_TASK_ID='calculate_fts' AIRFLOW_CTX_EXECUTION_DATE='2025-03-08T23:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='7' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-03-08T23:00:00+00:00'
[2025-04-22T17:14:27.231+0000] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2025-04-22T17:14:27.263+0000] {subprocess.py:75} INFO - Running command: ['/usr/bin/bash', '-c', '\n        python /opt/***/external_scripts/2_calculate_features.py 2025-03-09\n        ']
[2025-04-22T17:14:27.354+0000] {subprocess.py:86} INFO - Output:
[2025-04-22T17:14:36.844+0000] {subprocess.py:93} INFO - :: loading settings :: url = jar:file:/opt/spark/jars/ivy-2.5.1.jar!/org/apache/ivy/core/settings/ivysettings.xml
[2025-04-22T17:14:37.150+0000] {subprocess.py:93} INFO - Ivy Default Cache set to: /home/***/.ivy2/cache
[2025-04-22T17:14:37.151+0000] {subprocess.py:93} INFO - The jars for the packages stored in: /home/***/.ivy2/jars
[2025-04-22T17:14:37.164+0000] {subprocess.py:93} INFO - org.apache.spark#spark-sql-kafka-0-10_2.12 added as a dependency
[2025-04-22T17:14:37.167+0000] {subprocess.py:93} INFO - io.delta#delta-core_2.12 added as a dependency
[2025-04-22T17:14:37.167+0000] {subprocess.py:93} INFO - org.apache.hadoop#hadoop-aws added as a dependency
[2025-04-22T17:14:37.168+0000] {subprocess.py:93} INFO - com.amazonaws#aws-java-sdk-bundle added as a dependency
[2025-04-22T17:14:37.168+0000] {subprocess.py:93} INFO - org.postgresql#postgresql added as a dependency
[2025-04-22T17:14:37.168+0000] {subprocess.py:93} INFO - :: resolving dependencies :: org.apache.spark#spark-submit-parent-cc5500df-ba90-4779-9801-93d37a1f5e7b;1.0
[2025-04-22T17:14:37.168+0000] {subprocess.py:93} INFO - 	confs: [default]
[2025-04-22T17:14:37.463+0000] {subprocess.py:93} INFO - 	found org.apache.spark#spark-sql-kafka-0-10_2.12;3.4.1 in central
[2025-04-22T17:14:37.595+0000] {subprocess.py:93} INFO - 	found org.apache.spark#spark-token-provider-kafka-0-10_2.12;3.4.1 in central
[2025-04-22T17:14:37.679+0000] {subprocess.py:93} INFO - 	found org.apache.kafka#kafka-clients;3.3.2 in central
[2025-04-22T17:14:37.826+0000] {subprocess.py:93} INFO - 	found org.lz4#lz4-java;1.8.0 in central
[2025-04-22T17:14:37.953+0000] {subprocess.py:93} INFO - 	found org.xerial.snappy#snappy-java;1.1.10.1 in central
[2025-04-22T17:14:38.011+0000] {subprocess.py:93} INFO - 	found org.slf4j#slf4j-api;2.0.6 in central
[2025-04-22T17:14:38.079+0000] {subprocess.py:93} INFO - 	found org.apache.hadoop#hadoop-client-runtime;3.3.4 in central
[2025-04-22T17:14:38.116+0000] {subprocess.py:93} INFO - 	found org.apache.hadoop#hadoop-client-api;3.3.4 in central
[2025-04-22T17:14:38.173+0000] {subprocess.py:93} INFO - 	found commons-logging#commons-logging;1.1.3 in central
[2025-04-22T17:14:38.213+0000] {subprocess.py:93} INFO - 	found com.google.code.findbugs#jsr305;3.0.0 in central
[2025-04-22T17:14:38.250+0000] {subprocess.py:93} INFO - 	found org.apache.commons#commons-pool2;2.11.1 in central
[2025-04-22T17:14:38.289+0000] {subprocess.py:93} INFO - 	found io.delta#delta-core_2.12;2.4.0 in central
[2025-04-22T17:14:38.317+0000] {subprocess.py:93} INFO - 	found io.delta#delta-storage;2.4.0 in central
[2025-04-22T17:14:38.353+0000] {subprocess.py:93} INFO - 	found org.antlr#antlr4-runtime;4.9.3 in central
[2025-04-22T17:14:38.404+0000] {subprocess.py:93} INFO - 	found org.apache.hadoop#hadoop-aws;3.3.2 in central
[2025-04-22T17:14:38.431+0000] {subprocess.py:93} INFO - 	found org.wildfly.openssl#wildfly-openssl;1.0.7.Final in central
[2025-04-22T17:14:38.469+0000] {subprocess.py:93} INFO - 	found com.amazonaws#aws-java-sdk-bundle;1.12.261 in central
[2025-04-22T17:14:38.485+0000] {subprocess.py:93} INFO - 	found org.postgresql#postgresql;42.5.1 in central
[2025-04-22T17:14:38.520+0000] {subprocess.py:93} INFO - 	found org.checkerframework#checker-qual;3.5.0 in central
[2025-04-22T17:14:38.588+0000] {subprocess.py:93} INFO - :: resolution report :: resolve 1375ms :: artifacts dl 45ms
[2025-04-22T17:14:38.588+0000] {subprocess.py:93} INFO - 	:: modules in use:
[2025-04-22T17:14:38.589+0000] {subprocess.py:93} INFO - 	com.amazonaws#aws-java-sdk-bundle;1.12.261 from central in [default]
[2025-04-22T17:14:38.589+0000] {subprocess.py:93} INFO - 	com.google.code.findbugs#jsr305;3.0.0 from central in [default]
[2025-04-22T17:14:38.590+0000] {subprocess.py:93} INFO - 	commons-logging#commons-logging;1.1.3 from central in [default]
[2025-04-22T17:14:38.590+0000] {subprocess.py:93} INFO - 	io.delta#delta-core_2.12;2.4.0 from central in [default]
[2025-04-22T17:14:38.590+0000] {subprocess.py:93} INFO - 	io.delta#delta-storage;2.4.0 from central in [default]
[2025-04-22T17:14:38.590+0000] {subprocess.py:93} INFO - 	org.antlr#antlr4-runtime;4.9.3 from central in [default]
[2025-04-22T17:14:38.591+0000] {subprocess.py:93} INFO - 	org.apache.commons#commons-pool2;2.11.1 from central in [default]
[2025-04-22T17:14:38.591+0000] {subprocess.py:93} INFO - 	org.apache.hadoop#hadoop-aws;3.3.2 from central in [default]
[2025-04-22T17:14:38.591+0000] {subprocess.py:93} INFO - 	org.apache.hadoop#hadoop-client-api;3.3.4 from central in [default]
[2025-04-22T17:14:38.593+0000] {subprocess.py:93} INFO - 	org.apache.hadoop#hadoop-client-runtime;3.3.4 from central in [default]
[2025-04-22T17:14:38.593+0000] {subprocess.py:93} INFO - 	org.apache.kafka#kafka-clients;3.3.2 from central in [default]
[2025-04-22T17:14:38.594+0000] {subprocess.py:93} INFO - 	org.apache.spark#spark-sql-kafka-0-10_2.12;3.4.1 from central in [default]
[2025-04-22T17:14:38.594+0000] {subprocess.py:93} INFO - 	org.apache.spark#spark-token-provider-kafka-0-10_2.12;3.4.1 from central in [default]
[2025-04-22T17:14:38.594+0000] {subprocess.py:93} INFO - 	org.checkerframework#checker-qual;3.5.0 from central in [default]
[2025-04-22T17:14:38.594+0000] {subprocess.py:93} INFO - 	org.lz4#lz4-java;1.8.0 from central in [default]
[2025-04-22T17:14:38.594+0000] {subprocess.py:93} INFO - 	org.postgresql#postgresql;42.5.1 from central in [default]
[2025-04-22T17:14:38.594+0000] {subprocess.py:93} INFO - 	org.slf4j#slf4j-api;2.0.6 from central in [default]
[2025-04-22T17:14:38.594+0000] {subprocess.py:93} INFO - 	org.wildfly.openssl#wildfly-openssl;1.0.7.Final from central in [default]
[2025-04-22T17:14:38.594+0000] {subprocess.py:93} INFO - 	org.xerial.snappy#snappy-java;1.1.10.1 from central in [default]
[2025-04-22T17:14:38.594+0000] {subprocess.py:93} INFO - 	:: evicted modules:
[2025-04-22T17:14:38.594+0000] {subprocess.py:93} INFO - 	com.amazonaws#aws-java-sdk-bundle;1.11.1026 by [com.amazonaws#aws-java-sdk-bundle;1.12.261] in [default]
[2025-04-22T17:14:38.594+0000] {subprocess.py:93} INFO - 	---------------------------------------------------------------------
[2025-04-22T17:14:38.595+0000] {subprocess.py:93} INFO - 	|                  |            modules            ||   artifacts   |
[2025-04-22T17:14:38.595+0000] {subprocess.py:93} INFO - 	|       conf       | number| search|dwnlded|evicted|| number|dwnlded|
[2025-04-22T17:14:38.595+0000] {subprocess.py:93} INFO - 	---------------------------------------------------------------------
[2025-04-22T17:14:38.595+0000] {subprocess.py:93} INFO - 	|      default     |   20  |   0   |   0   |   1   ||   19  |   0   |
[2025-04-22T17:14:38.595+0000] {subprocess.py:93} INFO - 	---------------------------------------------------------------------
[2025-04-22T17:14:38.610+0000] {subprocess.py:93} INFO - :: retrieving :: org.apache.spark#spark-submit-parent-cc5500df-ba90-4779-9801-93d37a1f5e7b
[2025-04-22T17:14:38.610+0000] {subprocess.py:93} INFO - 	confs: [default]
[2025-04-22T17:14:38.637+0000] {subprocess.py:93} INFO - 	0 artifacts copied, 19 already retrieved (0kB/26ms)
[2025-04-22T17:14:39.227+0000] {subprocess.py:93} INFO - 25/04/22 17:14:39 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2025-04-22T17:14:39.616+0000] {subprocess.py:93} INFO - Setting default log level to "WARN".
[2025-04-22T17:14:39.617+0000] {subprocess.py:93} INFO - To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).
[2025-04-22T17:14:43.474+0000] {subprocess.py:93} INFO - 25/04/22 17:14:43 WARN Utils: Service 'SparkUI' could not bind on port 4040. Attempting port 4041.
[2025-04-22T17:15:08.527+0000] {subprocess.py:93} INFO - 25/04/22 17:15:08 ERROR Inbox: Ignoring error
[2025-04-22T17:15:08.579+0000] {subprocess.py:93} INFO - java.lang.NullPointerException
[2025-04-22T17:15:08.580+0000] {subprocess.py:93} INFO - 	at org.apache.spark.storage.BlockManagerMasterEndpoint.org$apache$spark$storage$BlockManagerMasterEndpoint$$register(BlockManagerMasterEndpoint.scala:600)
[2025-04-22T17:15:08.580+0000] {subprocess.py:93} INFO - 	at org.apache.spark.storage.BlockManagerMasterEndpoint$$anonfun$receiveAndReply$1.applyOrElse(BlockManagerMasterEndpoint.scala:123)
[2025-04-22T17:15:08.580+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.Inbox.$anonfun$process$1(Inbox.scala:103)
[2025-04-22T17:15:08.580+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.Inbox.safelyCall(Inbox.scala:213)
[2025-04-22T17:15:08.580+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.Inbox.process(Inbox.scala:100)
[2025-04-22T17:15:08.580+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.MessageLoop.org$apache$spark$rpc$netty$MessageLoop$$receiveLoop(MessageLoop.scala:75)
[2025-04-22T17:15:08.581+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.MessageLoop$$anon$1.run(MessageLoop.scala:41)
[2025-04-22T17:15:08.581+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[2025-04-22T17:15:08.581+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[2025-04-22T17:15:08.581+0000] {subprocess.py:93} INFO - 	at java.base/java.lang.Thread.run(Thread.java:829)
[2025-04-22T17:15:08.581+0000] {subprocess.py:93} INFO - 25/04/22 17:15:08 WARN Executor: Issue communicating with driver in heartbeater
[2025-04-22T17:15:08.582+0000] {subprocess.py:93} INFO - org.apache.spark.SparkException: Exception thrown in awaitResult:
[2025-04-22T17:15:08.582+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.ThreadUtils$.awaitResult(ThreadUtils.scala:322)
[2025-04-22T17:15:08.582+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.RpcTimeout.awaitResult(RpcTimeout.scala:75)
[2025-04-22T17:15:08.582+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.RpcEndpointRef.askSync(RpcEndpointRef.scala:101)
[2025-04-22T17:15:08.582+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.RpcEndpointRef.askSync(RpcEndpointRef.scala:85)
[2025-04-22T17:15:08.582+0000] {subprocess.py:93} INFO - 	at org.apache.spark.storage.BlockManagerMaster.registerBlockManager(BlockManagerMaster.scala:80)
[2025-04-22T17:15:08.582+0000] {subprocess.py:93} INFO - 	at org.apache.spark.storage.BlockManager.reregister(BlockManager.scala:641)
[2025-04-22T17:15:08.582+0000] {subprocess.py:93} INFO - 	at org.apache.spark.executor.Executor.reportHeartBeat(Executor.scala:1111)
[2025-04-22T17:15:08.582+0000] {subprocess.py:93} INFO - 	at org.apache.spark.executor.Executor.$anonfun$heartbeater$1(Executor.scala:244)
[2025-04-22T17:15:08.582+0000] {subprocess.py:93} INFO - 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[2025-04-22T17:15:08.583+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:2088)
[2025-04-22T17:15:08.583+0000] {subprocess.py:93} INFO - 	at org.apache.spark.Heartbeater$$anon$1.run(Heartbeater.scala:46)
[2025-04-22T17:15:08.583+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
[2025-04-22T17:15:08.583+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
[2025-04-22T17:15:08.583+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
[2025-04-22T17:15:08.583+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[2025-04-22T17:15:08.583+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[2025-04-22T17:15:08.583+0000] {subprocess.py:93} INFO - 	at java.base/java.lang.Thread.run(Thread.java:829)
[2025-04-22T17:15:08.583+0000] {subprocess.py:93} INFO - Caused by: java.lang.NullPointerException
[2025-04-22T17:15:08.583+0000] {subprocess.py:93} INFO - 	at org.apache.spark.storage.BlockManagerMasterEndpoint.org$apache$spark$storage$BlockManagerMasterEndpoint$$register(BlockManagerMasterEndpoint.scala:600)
[2025-04-22T17:15:08.583+0000] {subprocess.py:93} INFO - 	at org.apache.spark.storage.BlockManagerMasterEndpoint$$anonfun$receiveAndReply$1.applyOrElse(BlockManagerMasterEndpoint.scala:123)
[2025-04-22T17:15:08.583+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.Inbox.$anonfun$process$1(Inbox.scala:103)
[2025-04-22T17:15:08.584+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.Inbox.safelyCall(Inbox.scala:213)
[2025-04-22T17:15:08.584+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.Inbox.process(Inbox.scala:100)
[2025-04-22T17:15:08.584+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.MessageLoop.org$apache$spark$rpc$netty$MessageLoop$$receiveLoop(MessageLoop.scala:75)
[2025-04-22T17:15:08.584+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.MessageLoop$$anon$1.run(MessageLoop.scala:41)
[2025-04-22T17:15:08.586+0000] {subprocess.py:93} INFO - 	... 3 more
[2025-04-22T17:15:18.181+0000] {subprocess.py:93} INFO - 25/04/22 17:15:18 WARN Executor: Issue communicating with driver in heartbeater
[2025-04-22T17:15:18.181+0000] {subprocess.py:93} INFO - org.apache.spark.SparkException: Exception thrown in awaitResult:
[2025-04-22T17:15:18.181+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.ThreadUtils$.awaitResult(ThreadUtils.scala:322)
[2025-04-22T17:15:18.182+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.RpcTimeout.awaitResult(RpcTimeout.scala:75)
[2025-04-22T17:15:18.182+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.RpcEndpointRef.askSync(RpcEndpointRef.scala:101)
[2025-04-22T17:15:18.182+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.RpcEndpointRef.askSync(RpcEndpointRef.scala:85)
[2025-04-22T17:15:18.182+0000] {subprocess.py:93} INFO - 	at org.apache.spark.storage.BlockManagerMaster.registerBlockManager(BlockManagerMaster.scala:80)
[2025-04-22T17:15:18.182+0000] {subprocess.py:93} INFO - 	at org.apache.spark.storage.BlockManager.reregister(BlockManager.scala:641)
[2025-04-22T17:15:18.183+0000] {subprocess.py:93} INFO - 	at org.apache.spark.executor.Executor.reportHeartBeat(Executor.scala:1111)
[2025-04-22T17:15:18.183+0000] {subprocess.py:93} INFO - 	at org.apache.spark.executor.Executor.$anonfun$heartbeater$1(Executor.scala:244)
[2025-04-22T17:15:18.183+0000] {subprocess.py:93} INFO - 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[2025-04-22T17:15:18.183+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.Utils$.logUncaughtExceptions(Utils.scala:2088)
[2025-04-22T17:15:18.183+0000] {subprocess.py:93} INFO - 	at org.apache.spark.Heartbeater$$anon$1.run(Heartbeater.scala:46)
[2025-04-22T17:15:18.184+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
[2025-04-22T17:15:18.184+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
[2025-04-22T17:15:18.184+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
[2025-04-22T17:15:18.184+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[2025-04-22T17:15:18.184+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[2025-04-22T17:15:18.185+0000] {subprocess.py:93} INFO - 	at java.base/java.lang.Thread.run(Thread.java:829)
[2025-04-22T17:15:18.185+0000] {subprocess.py:93} INFO - Caused by: org.apache.spark.SparkException: Exception thrown in awaitResult:
[2025-04-22T17:15:18.185+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.ThreadUtils$.awaitResult(ThreadUtils.scala:322)
[2025-04-22T17:15:18.185+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.RpcTimeout.awaitResult(RpcTimeout.scala:75)
[2025-04-22T17:15:18.186+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.RpcEnv.setupEndpointRefByURI(RpcEnv.scala:102)
[2025-04-22T17:15:18.206+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.RpcEnv.setupEndpointRef(RpcEnv.scala:110)
[2025-04-22T17:15:18.207+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.RpcUtils$.makeDriverRef(RpcUtils.scala:36)
[2025-04-22T17:15:18.215+0000] {subprocess.py:93} INFO - 	at org.apache.spark.storage.BlockManagerMasterEndpoint.driverEndpoint$lzycompute(BlockManagerMasterEndpoint.scala:117)
[2025-04-22T17:15:18.215+0000] {subprocess.py:93} INFO - 	at org.apache.spark.storage.BlockManagerMasterEndpoint.org$apache$spark$storage$BlockManagerMasterEndpoint$$driverEndpoint(BlockManagerMasterEndpoint.scala:116)
[2025-04-22T17:15:18.216+0000] {subprocess.py:93} INFO - 	at org.apache.spark.storage.BlockManagerMasterEndpoint.isExecutorAlive$lzycompute$1(BlockManagerMasterEndpoint.scala:611)
[2025-04-22T17:15:18.216+0000] {subprocess.py:93} INFO - 	at org.apache.spark.storage.BlockManagerMasterEndpoint.isExecutorAlive$1(BlockManagerMasterEndpoint.scala:610)
[2025-04-22T17:15:18.219+0000] {subprocess.py:93} INFO - 	at org.apache.spark.storage.BlockManagerMasterEndpoint.org$apache$spark$storage$BlockManagerMasterEndpoint$$register(BlockManagerMasterEndpoint.scala:648)
[2025-04-22T17:15:18.219+0000] {subprocess.py:93} INFO - 	at org.apache.spark.storage.BlockManagerMasterEndpoint$$anonfun$receiveAndReply$1.applyOrElse(BlockManagerMasterEndpoint.scala:123)
[2025-04-22T17:15:18.228+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.Inbox.$anonfun$process$1(Inbox.scala:103)
[2025-04-22T17:15:18.228+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.Inbox.safelyCall(Inbox.scala:213)
[2025-04-22T17:15:18.228+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.Inbox.process(Inbox.scala:100)
[2025-04-22T17:15:18.229+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.MessageLoop.org$apache$spark$rpc$netty$MessageLoop$$receiveLoop(MessageLoop.scala:75)
[2025-04-22T17:15:18.229+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.MessageLoop$$anon$1.run(MessageLoop.scala:41)
[2025-04-22T17:15:18.239+0000] {subprocess.py:93} INFO - 	... 3 more
[2025-04-22T17:15:18.239+0000] {subprocess.py:93} INFO - Caused by: org.apache.spark.rpc.RpcEndpointNotFoundException: Cannot find endpoint: spark://CoarseGrainedScheduler@45b38884c639:44807
[2025-04-22T17:15:18.239+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.NettyRpcEnv.$anonfun$asyncSetupEndpointRefByURI$1(NettyRpcEnv.scala:148)
[2025-04-22T17:15:18.239+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.NettyRpcEnv.$anonfun$asyncSetupEndpointRefByURI$1$adapted(NettyRpcEnv.scala:144)
[2025-04-22T17:15:18.240+0000] {subprocess.py:93} INFO - 	at scala.concurrent.Future.$anonfun$flatMap$1(Future.scala:307)
[2025-04-22T17:15:18.240+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:41)
[2025-04-22T17:15:18.240+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
[2025-04-22T17:15:18.240+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.ThreadUtils$$anon$1.execute(ThreadUtils.scala:99)
[2025-04-22T17:15:18.240+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.ExecutionContextImpl$$anon$4.execute(ExecutionContextImpl.scala:138)
[2025-04-22T17:15:18.240+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:72)
[2025-04-22T17:15:18.240+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1(Promise.scala:288)
[2025-04-22T17:15:18.240+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1$adapted(Promise.scala:288)
[2025-04-22T17:15:18.241+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:288)
[2025-04-22T17:15:18.259+0000] {subprocess.py:93} INFO - 	at scala.concurrent.Promise.complete(Promise.scala:53)
[2025-04-22T17:15:18.260+0000] {subprocess.py:93} INFO - 	at scala.concurrent.Promise.complete$(Promise.scala:52)
[2025-04-22T17:15:18.260+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:187)
[2025-04-22T17:15:18.260+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
[2025-04-22T17:15:18.260+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
[2025-04-22T17:15:18.260+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.ThreadUtils$$anon$1.execute(ThreadUtils.scala:99)
[2025-04-22T17:15:18.261+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.ExecutionContextImpl$$anon$4.execute(ExecutionContextImpl.scala:138)
[2025-04-22T17:15:18.261+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:72)
[2025-04-22T17:15:18.261+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1(Promise.scala:288)
[2025-04-22T17:15:18.261+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1$adapted(Promise.scala:288)
[2025-04-22T17:15:18.261+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:288)
[2025-04-22T17:15:18.261+0000] {subprocess.py:93} INFO - 	at scala.concurrent.Promise.complete(Promise.scala:53)
[2025-04-22T17:15:18.262+0000] {subprocess.py:93} INFO - 	at scala.concurrent.Promise.complete$(Promise.scala:52)
[2025-04-22T17:15:18.286+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:187)
[2025-04-22T17:15:18.289+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
[2025-04-22T17:15:18.293+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
[2025-04-22T17:15:18.293+0000] {subprocess.py:93} INFO - 	at scala.concurrent.BatchingExecutor$Batch.processBatch$1(BatchingExecutor.scala:67)
[2025-04-22T17:15:18.293+0000] {subprocess.py:93} INFO - 	at scala.concurrent.BatchingExecutor$Batch.$anonfun$run$1(BatchingExecutor.scala:82)
[2025-04-22T17:15:18.294+0000] {subprocess.py:93} INFO - 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[2025-04-22T17:15:18.294+0000] {subprocess.py:93} INFO - 	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:85)
[2025-04-22T17:15:18.294+0000] {subprocess.py:93} INFO - 	at scala.concurrent.BatchingExecutor$Batch.run(BatchingExecutor.scala:59)
[2025-04-22T17:15:18.294+0000] {subprocess.py:93} INFO - 	at scala.concurrent.Future$InternalCallbackExecutor$.unbatchedExecute(Future.scala:875)
[2025-04-22T17:15:18.294+0000] {subprocess.py:93} INFO - 	at scala.concurrent.BatchingExecutor.execute(BatchingExecutor.scala:110)
[2025-04-22T17:15:18.294+0000] {subprocess.py:93} INFO - 	at scala.concurrent.BatchingExecutor.execute$(BatchingExecutor.scala:107)
[2025-04-22T17:15:18.294+0000] {subprocess.py:93} INFO - 	at scala.concurrent.Future$InternalCallbackExecutor$.execute(Future.scala:873)
[2025-04-22T17:15:18.295+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:72)
[2025-04-22T17:15:18.295+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1(Promise.scala:288)
[2025-04-22T17:15:18.295+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1$adapted(Promise.scala:288)
[2025-04-22T17:15:18.295+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:288)
[2025-04-22T17:15:18.295+0000] {subprocess.py:93} INFO - 	at scala.concurrent.Promise.trySuccess(Promise.scala:94)
[2025-04-22T17:15:18.296+0000] {subprocess.py:93} INFO - 	at scala.concurrent.Promise.trySuccess$(Promise.scala:94)
[2025-04-22T17:15:18.296+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise$DefaultPromise.trySuccess(Promise.scala:187)
[2025-04-22T17:15:18.296+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.NettyRpcEnv.onSuccess$1(NettyRpcEnv.scala:225)
[2025-04-22T17:15:18.296+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.NettyRpcEnv.$anonfun$askAbortable$5(NettyRpcEnv.scala:239)
[2025-04-22T17:15:18.297+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.NettyRpcEnv.$anonfun$askAbortable$5$adapted(NettyRpcEnv.scala:238)
[2025-04-22T17:15:18.297+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
[2025-04-22T17:15:18.297+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.ThreadUtils$$anon$1.execute(ThreadUtils.scala:99)
[2025-04-22T17:15:18.297+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.ExecutionContextImpl$$anon$4.execute(ExecutionContextImpl.scala:138)
[2025-04-22T17:15:18.297+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:72)
[2025-04-22T17:15:18.297+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1(Promise.scala:288)
[2025-04-22T17:15:18.297+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1$adapted(Promise.scala:288)
[2025-04-22T17:15:18.298+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:288)
[2025-04-22T17:15:18.298+0000] {subprocess.py:93} INFO - 	at scala.concurrent.Promise.complete(Promise.scala:53)
[2025-04-22T17:15:18.298+0000] {subprocess.py:93} INFO - 	at scala.concurrent.Promise.complete$(Promise.scala:52)
[2025-04-22T17:15:18.298+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:187)
[2025-04-22T17:15:18.298+0000] {subprocess.py:93} INFO - 	at scala.concurrent.Promise.success(Promise.scala:86)
[2025-04-22T17:15:18.298+0000] {subprocess.py:93} INFO - 	at scala.concurrent.Promise.success$(Promise.scala:86)
[2025-04-22T17:15:18.298+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise$DefaultPromise.success(Promise.scala:187)
[2025-04-22T17:15:18.299+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.LocalNettyRpcCallContext.send(NettyRpcCallContext.scala:50)
[2025-04-22T17:15:18.299+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.NettyRpcCallContext.reply(NettyRpcCallContext.scala:32)
[2025-04-22T17:15:18.299+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.RpcEndpointVerifier$$anonfun$receiveAndReply$1.applyOrElse(RpcEndpointVerifier.scala:31)
[2025-04-22T17:15:18.299+0000] {subprocess.py:93} INFO - 	... 8 more
[2025-04-22T17:15:18.299+0000] {subprocess.py:93} INFO - 25/04/22 17:15:18 ERROR Inbox: Ignoring error
[2025-04-22T17:15:18.299+0000] {subprocess.py:93} INFO - org.apache.spark.SparkException: Exception thrown in awaitResult:
[2025-04-22T17:15:18.299+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.ThreadUtils$.awaitResult(ThreadUtils.scala:322)
[2025-04-22T17:15:18.299+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.RpcTimeout.awaitResult(RpcTimeout.scala:75)
[2025-04-22T17:15:18.300+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.RpcEnv.setupEndpointRefByURI(RpcEnv.scala:102)
[2025-04-22T17:15:18.300+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.RpcEnv.setupEndpointRef(RpcEnv.scala:110)
[2025-04-22T17:15:18.300+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.RpcUtils$.makeDriverRef(RpcUtils.scala:36)
[2025-04-22T17:15:18.300+0000] {subprocess.py:93} INFO - 	at org.apache.spark.storage.BlockManagerMasterEndpoint.driverEndpoint$lzycompute(BlockManagerMasterEndpoint.scala:117)
[2025-04-22T17:15:18.300+0000] {subprocess.py:93} INFO - 	at org.apache.spark.storage.BlockManagerMasterEndpoint.org$apache$spark$storage$BlockManagerMasterEndpoint$$driverEndpoint(BlockManagerMasterEndpoint.scala:116)
[2025-04-22T17:15:18.300+0000] {subprocess.py:93} INFO - 	at org.apache.spark.storage.BlockManagerMasterEndpoint.isExecutorAlive$lzycompute$1(BlockManagerMasterEndpoint.scala:611)
[2025-04-22T17:15:18.300+0000] {subprocess.py:93} INFO - 	at org.apache.spark.storage.BlockManagerMasterEndpoint.isExecutorAlive$1(BlockManagerMasterEndpoint.scala:610)
[2025-04-22T17:15:18.300+0000] {subprocess.py:93} INFO - 	at org.apache.spark.storage.BlockManagerMasterEndpoint.org$apache$spark$storage$BlockManagerMasterEndpoint$$register(BlockManagerMasterEndpoint.scala:648)
[2025-04-22T17:15:18.301+0000] {subprocess.py:93} INFO - 	at org.apache.spark.storage.BlockManagerMasterEndpoint$$anonfun$receiveAndReply$1.applyOrElse(BlockManagerMasterEndpoint.scala:123)
[2025-04-22T17:15:18.301+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.Inbox.$anonfun$process$1(Inbox.scala:103)
[2025-04-22T17:15:18.301+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.Inbox.safelyCall(Inbox.scala:213)
[2025-04-22T17:15:18.301+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.Inbox.process(Inbox.scala:100)
[2025-04-22T17:15:18.301+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.MessageLoop.org$apache$spark$rpc$netty$MessageLoop$$receiveLoop(MessageLoop.scala:75)
[2025-04-22T17:15:18.301+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.MessageLoop$$anon$1.run(MessageLoop.scala:41)
[2025-04-22T17:15:18.302+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[2025-04-22T17:15:18.302+0000] {subprocess.py:93} INFO - 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[2025-04-22T17:15:18.302+0000] {subprocess.py:93} INFO - 	at java.base/java.lang.Thread.run(Thread.java:829)
[2025-04-22T17:15:18.302+0000] {subprocess.py:93} INFO - Caused by: org.apache.spark.rpc.RpcEndpointNotFoundException: Cannot find endpoint: spark://CoarseGrainedScheduler@45b38884c639:44807
[2025-04-22T17:15:18.302+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.NettyRpcEnv.$anonfun$asyncSetupEndpointRefByURI$1(NettyRpcEnv.scala:148)
[2025-04-22T17:15:18.302+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.NettyRpcEnv.$anonfun$asyncSetupEndpointRefByURI$1$adapted(NettyRpcEnv.scala:144)
[2025-04-22T17:15:18.302+0000] {subprocess.py:93} INFO - 	at scala.concurrent.Future.$anonfun$flatMap$1(Future.scala:307)
[2025-04-22T17:15:18.302+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise.$anonfun$transformWith$1(Promise.scala:41)
[2025-04-22T17:15:18.302+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
[2025-04-22T17:15:18.303+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.ThreadUtils$$anon$1.execute(ThreadUtils.scala:99)
[2025-04-22T17:15:18.303+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.ExecutionContextImpl$$anon$4.execute(ExecutionContextImpl.scala:138)
[2025-04-22T17:15:18.303+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:72)
[2025-04-22T17:15:18.303+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1(Promise.scala:288)
[2025-04-22T17:15:18.303+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1$adapted(Promise.scala:288)
[2025-04-22T17:15:18.303+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:288)
[2025-04-22T17:15:18.303+0000] {subprocess.py:93} INFO - 	at scala.concurrent.Promise.complete(Promise.scala:53)
[2025-04-22T17:15:18.303+0000] {subprocess.py:93} INFO - 	at scala.concurrent.Promise.complete$(Promise.scala:52)
[2025-04-22T17:15:18.303+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:187)
[2025-04-22T17:15:18.303+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
[2025-04-22T17:15:18.303+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
[2025-04-22T17:15:18.304+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.ThreadUtils$$anon$1.execute(ThreadUtils.scala:99)
[2025-04-22T17:15:18.304+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.ExecutionContextImpl$$anon$4.execute(ExecutionContextImpl.scala:138)
[2025-04-22T17:15:18.304+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:72)
[2025-04-22T17:15:18.304+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1(Promise.scala:288)
[2025-04-22T17:15:18.304+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1$adapted(Promise.scala:288)
[2025-04-22T17:15:18.304+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:288)
[2025-04-22T17:15:18.304+0000] {subprocess.py:93} INFO - 	at scala.concurrent.Promise.complete(Promise.scala:53)
[2025-04-22T17:15:18.304+0000] {subprocess.py:93} INFO - 	at scala.concurrent.Promise.complete$(Promise.scala:52)
[2025-04-22T17:15:18.304+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:187)
[2025-04-22T17:15:18.304+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
[2025-04-22T17:15:18.304+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
[2025-04-22T17:15:18.305+0000] {subprocess.py:93} INFO - 	at scala.concurrent.BatchingExecutor$Batch.processBatch$1(BatchingExecutor.scala:67)
[2025-04-22T17:15:18.305+0000] {subprocess.py:93} INFO - 	at scala.concurrent.BatchingExecutor$Batch.$anonfun$run$1(BatchingExecutor.scala:82)
[2025-04-22T17:15:18.305+0000] {subprocess.py:93} INFO - 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[2025-04-22T17:15:18.305+0000] {subprocess.py:93} INFO - 	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:85)
[2025-04-22T17:15:18.305+0000] {subprocess.py:93} INFO - 	at scala.concurrent.BatchingExecutor$Batch.run(BatchingExecutor.scala:59)
[2025-04-22T17:15:18.305+0000] {subprocess.py:93} INFO - 	at scala.concurrent.Future$InternalCallbackExecutor$.unbatchedExecute(Future.scala:875)
[2025-04-22T17:15:18.305+0000] {subprocess.py:93} INFO - 	at scala.concurrent.BatchingExecutor.execute(BatchingExecutor.scala:110)
[2025-04-22T17:15:18.305+0000] {subprocess.py:93} INFO - 	at scala.concurrent.BatchingExecutor.execute$(BatchingExecutor.scala:107)
[2025-04-22T17:15:18.305+0000] {subprocess.py:93} INFO - 	at scala.concurrent.Future$InternalCallbackExecutor$.execute(Future.scala:873)
[2025-04-22T17:15:18.305+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:72)
[2025-04-22T17:15:18.305+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1(Promise.scala:288)
[2025-04-22T17:15:18.305+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1$adapted(Promise.scala:288)
[2025-04-22T17:15:18.306+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:288)
[2025-04-22T17:15:18.306+0000] {subprocess.py:93} INFO - 	at scala.concurrent.Promise.trySuccess(Promise.scala:94)
[2025-04-22T17:15:18.306+0000] {subprocess.py:93} INFO - 	at scala.concurrent.Promise.trySuccess$(Promise.scala:94)
[2025-04-22T17:15:18.306+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise$DefaultPromise.trySuccess(Promise.scala:187)
[2025-04-22T17:15:18.306+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.NettyRpcEnv.onSuccess$1(NettyRpcEnv.scala:225)
[2025-04-22T17:15:18.306+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.NettyRpcEnv.$anonfun$askAbortable$5(NettyRpcEnv.scala:239)
[2025-04-22T17:15:18.306+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.NettyRpcEnv.$anonfun$askAbortable$5$adapted(NettyRpcEnv.scala:238)
[2025-04-22T17:15:18.306+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
[2025-04-22T17:15:18.306+0000] {subprocess.py:93} INFO - 	at org.apache.spark.util.ThreadUtils$$anon$1.execute(ThreadUtils.scala:99)
[2025-04-22T17:15:18.306+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.ExecutionContextImpl$$anon$4.execute(ExecutionContextImpl.scala:138)
[2025-04-22T17:15:18.306+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.CallbackRunnable.executeWithValue(Promise.scala:72)
[2025-04-22T17:15:18.307+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1(Promise.scala:288)
[2025-04-22T17:15:18.307+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise$DefaultPromise.$anonfun$tryComplete$1$adapted(Promise.scala:288)
[2025-04-22T17:15:18.307+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise$DefaultPromise.tryComplete(Promise.scala:288)
[2025-04-22T17:15:18.307+0000] {subprocess.py:93} INFO - 	at scala.concurrent.Promise.complete(Promise.scala:53)
[2025-04-22T17:15:18.307+0000] {subprocess.py:93} INFO - 	at scala.concurrent.Promise.complete$(Promise.scala:52)
[2025-04-22T17:15:18.307+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise$DefaultPromise.complete(Promise.scala:187)
[2025-04-22T17:15:18.307+0000] {subprocess.py:93} INFO - 	at scala.concurrent.Promise.success(Promise.scala:86)
[2025-04-22T17:15:18.307+0000] {subprocess.py:93} INFO - 	at scala.concurrent.Promise.success$(Promise.scala:86)
[2025-04-22T17:15:18.307+0000] {subprocess.py:93} INFO - 	at scala.concurrent.impl.Promise$DefaultPromise.success(Promise.scala:187)
[2025-04-22T17:15:18.307+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.LocalNettyRpcCallContext.send(NettyRpcCallContext.scala:50)
[2025-04-22T17:15:18.307+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.NettyRpcCallContext.reply(NettyRpcCallContext.scala:32)
[2025-04-22T17:15:18.308+0000] {subprocess.py:93} INFO - 	at org.apache.spark.rpc.netty.RpcEndpointVerifier$$anonfun$receiveAndReply$1.applyOrElse(RpcEndpointVerifier.scala:31)
[2025-04-22T17:15:18.308+0000] {subprocess.py:93} INFO - 	... 8 more
[2025-04-22T17:15:20.409+0000] {subprocess.py:93} INFO - 25/04/22 17:15:20 WARN SQLConf: The SQL config 'spark.sql.execution.arrow.fallback.enabled' has been deprecated in Spark v3.0 and may be removed in the future. Use 'spark.sql.execution.arrow.pyspark.fallback.enabled' instead of it.
[2025-04-22T17:15:20.412+0000] {subprocess.py:93} INFO - 25/04/22 17:15:20 WARN SQLConf: The SQL config 'spark.sql.execution.arrow.fallback.enabled' has been deprecated in Spark v3.0 and may be removed in the future. Use 'spark.sql.execution.arrow.pyspark.fallback.enabled' instead of it.
[2025-04-22T17:15:20.413+0000] {subprocess.py:93} INFO - 25/04/22 17:15:20 WARN SQLConf: The SQL config 'spark.sql.execution.arrow.fallback.enabled' has been deprecated in Spark v3.0 and may be removed in the future. Use 'spark.sql.execution.arrow.pyspark.fallback.enabled' instead of it.
[2025-04-22T17:15:32.551+0000] {subprocess.py:93} INFO - 25/04/22 17:15:32 WARN AdminClientConfig: These configurations '[key.deserializer, value.deserializer, enable.auto.commit, max.poll.records, auto.offset.reset]' were supplied but are not used yet.
[2025-04-22T17:15:54.388+0000] {subprocess.py:93} INFO - [Stage 0:>                                                          (0 + 0) / 1][Stage 0:>                                                          (0 + 1) / 1]                                                                                25/04/22 17:15:54 WARN AdminClientConfig: These configurations '[key.deserializer, value.deserializer, enable.auto.commit, max.poll.records, auto.offset.reset]' were supplied but are not used yet.
[2025-04-22T17:15:54.618+0000] {subprocess.py:93} INFO - 25/04/22 17:15:54 WARN AdminClientConfig: These configurations '[key.deserializer, value.deserializer, enable.auto.commit, max.poll.records, auto.offset.reset]' were supplied but are not used yet.
[2025-04-22T17:16:11.736+0000] {subprocess.py:93} INFO - [Stage 1:>                                                          (0 + 1) / 1][Stage 1:>                  (0 + 1) / 1][Stage 2:>                  (0 + 1) / 1][Stage 1:>                                                          (0 + 1) / 1]                                                                                25/04/22 17:16:11 WARN AdminClientConfig: These configurations '[key.deserializer, value.deserializer, enable.auto.commit, max.poll.records, auto.offset.reset]' were supplied but are not used yet.
[2025-04-22T17:16:11.926+0000] {subprocess.py:93} INFO - 25/04/22 17:16:11 WARN AdminClientConfig: These configurations '[key.deserializer, value.deserializer, enable.auto.commit, max.poll.records, auto.offset.reset]' were supplied but are not used yet.
[2025-04-22T17:16:29.680+0000] {subprocess.py:93} INFO - [Stage 5:>                  (0 + 1) / 1][Stage 6:>                  (0 + 1) / 1][Stage 5:>                                                          (0 + 1) / 1][Stage 8:>                                                          (0 + 1) / 1]                                                                                25/04/22 17:16:29 WARN AdminClientConfig: These configurations '[key.deserializer, value.deserializer, enable.auto.commit, max.poll.records, auto.offset.reset]' were supplied but are not used yet.
[2025-04-22T17:16:30.857+0000] {subprocess.py:93} INFO - 25/04/22 17:16:30 WARN AdminClientConfig: These configurations '[key.deserializer, value.deserializer, enable.auto.commit, max.poll.records, auto.offset.reset]' were supplied but are not used yet.
[2025-04-22T17:16:59.573+0000] {subprocess.py:93} INFO - [Stage 16:>                 (0 + 1) / 1][Stage 17:>                 (0 + 1) / 1][Stage 16:>                                                         (0 + 1) / 1][Stage 19:>                                                         (0 + 5) / 5][Stage 19:===========>                                              (1 + 4) / 5][Stage 22:>                                                         (0 + 1) / 1]                                                                                25/04/22 17:16:59 WARN MetricsConfig: Cannot locate configuration: tried hadoop-metrics2-s3a-file-system.properties,hadoop-metrics2.properties
[2025-04-22T17:17:27.150+0000] {subprocess.py:93} INFO - ERROR:root:Exception while sending command.
[2025-04-22T17:17:27.166+0000] {subprocess.py:93} INFO - Traceback (most recent call last):
[2025-04-22T17:17:27.168+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/py4j-0.10.9.7-src.zip/py4j/clientserver.py", line 516, in send_command
[2025-04-22T17:17:27.168+0000] {subprocess.py:93} INFO -     raise Py4JNetworkError("Answer from Java side is empty")
[2025-04-22T17:17:27.169+0000] {subprocess.py:93} INFO - py4j.protocol.Py4JNetworkError: Answer from Java side is empty
[2025-04-22T17:17:27.169+0000] {subprocess.py:93} INFO - 
[2025-04-22T17:17:27.169+0000] {subprocess.py:93} INFO - During handling of the above exception, another exception occurred:
[2025-04-22T17:17:27.169+0000] {subprocess.py:93} INFO - 
[2025-04-22T17:17:27.170+0000] {subprocess.py:93} INFO - Traceback (most recent call last):
[2025-04-22T17:17:27.170+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/py4j-0.10.9.7-src.zip/py4j/java_gateway.py", line 1038, in send_command
[2025-04-22T17:17:27.170+0000] {subprocess.py:93} INFO -     response = connection.send_command(command)
[2025-04-22T17:17:27.170+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/py4j-0.10.9.7-src.zip/py4j/clientserver.py", line 539, in send_command
[2025-04-22T17:17:27.170+0000] {subprocess.py:93} INFO -     raise Py4JNetworkError(
[2025-04-22T17:17:27.170+0000] {subprocess.py:93} INFO - py4j.protocol.Py4JNetworkError: Error while sending or receiving
[2025-04-22T17:17:27.170+0000] {subprocess.py:93} INFO - 2025-03-02 2025-03-09
[2025-04-22T17:17:27.170+0000] {subprocess.py:93} INFO - Spark session initialized
[2025-04-22T17:17:27.171+0000] {subprocess.py:93} INFO - +------+--------------------+--------------------+---+-------------+-----------+
[2025-04-22T17:17:27.171+0000] {subprocess.py:93} INFO - |before|               after|              source| op|        ts_ms|transaction|
[2025-04-22T17:17:27.171+0000] {subprocess.py:93} INFO - +------+--------------------+--------------------+---+-------------+-----------+
[2025-04-22T17:17:27.171+0000] {subprocess.py:93} INFO - |  null|{1, user_000461, ...|{2.3.4.Final, pos...|  c|1745341450710|       null|
[2025-04-22T17:17:27.171+0000] {subprocess.py:93} INFO - |  null|{2, user_000113, ...|{2.3.4.Final, pos...|  c|1745341450735|       null|
[2025-04-22T17:17:27.171+0000] {subprocess.py:93} INFO - |  null|{3, user_000286, ...|{2.3.4.Final, pos...|  c|1745341450737|       null|
[2025-04-22T17:17:27.171+0000] {subprocess.py:93} INFO - |  null|{4, user_000402, ...|{2.3.4.Final, pos...|  c|1745341450738|       null|
[2025-04-22T17:17:27.171+0000] {subprocess.py:93} INFO - |  null|{5, user_000126, ...|{2.3.4.Final, pos...|  c|1745341450739|       null|
[2025-04-22T17:17:27.171+0000] {subprocess.py:93} INFO - +------+--------------------+--------------------+---+-------------+-----------+
[2025-04-22T17:17:27.172+0000] {subprocess.py:93} INFO - only showing top 5 rows
[2025-04-22T17:17:27.172+0000] {subprocess.py:93} INFO - 
[2025-04-22T17:17:27.172+0000] {subprocess.py:93} INFO - +---+-----------+--------------+------------------+-----------+---------------+-------------------+----------+
[2025-04-22T17:17:27.172+0000] {subprocess.py:93} INFO - | op|    User ID|Transaction ID|            Amount|     Vendor|        Sources|               Time|      date|
[2025-04-22T17:17:27.172+0000] {subprocess.py:93} INFO - +---+-----------+--------------+------------------+-----------+---------------+-------------------+----------+
[2025-04-22T17:17:27.172+0000] {subprocess.py:93} INFO - |  c|user_000284|  103511393417| 236.2428420919406|     Travel|    Credit Card|2025-03-02 12:59:03|2025-03-02|
[2025-04-22T17:17:27.172+0000] {subprocess.py:93} INFO - |  c|user_000461|  887242918791| 691.0975048741923|Electronics|Current Account|2025-03-02 16:53:36|2025-03-02|
[2025-04-22T17:17:27.172+0000] {subprocess.py:93} INFO - |  c|user_000250|  922032812744| 883.1256511491597| Restaurant|Current Account|2025-03-02 17:55:15|2025-03-02|
[2025-04-22T17:17:27.173+0000] {subprocess.py:93} INFO - |  c|user_000469|  646942374853| 716.2719387687761|   Clothing|Current Account|2025-03-02 16:32:22|2025-03-02|
[2025-04-22T17:17:27.173+0000] {subprocess.py:93} INFO - |  c|user_000122|  822385246258|259.96542928948867|  Education|Current Account|2025-03-02 14:49:15|2025-03-02|
[2025-04-22T17:17:27.173+0000] {subprocess.py:93} INFO - +---+-----------+--------------+------------------+-----------+---------------+-------------------+----------+
[2025-04-22T17:17:27.173+0000] {subprocess.py:93} INFO - only showing top 5 rows
[2025-04-22T17:17:27.173+0000] {subprocess.py:93} INFO - 
[2025-04-22T17:17:27.173+0000] {subprocess.py:93} INFO - +-----------+--------------------+------------------+------------------+------------------+-----------------+---------------+---------------+----------+
[2025-04-22T17:17:27.173+0000] {subprocess.py:93} INFO - |    user_id|num_transactions_l1w|  total_amount_l1w|    avg_amount_l1w|    min_amount_l1w|   max_amount_l1w|num_vendors_l1w|num_sources_l1w|      date|
[2025-04-22T17:17:27.173+0000] {subprocess.py:93} INFO - +-----------+--------------------+------------------+------------------+------------------+-----------------+---------------+---------------+----------+
[2025-04-22T17:17:27.174+0000] {subprocess.py:93} INFO - |user_000424|                  89| 43005.37925015618|483.20650842872107|16.521521963895324|977.4093263719208|             14|              3|2025-03-09|
[2025-04-22T17:17:27.174+0000] {subprocess.py:93} INFO - |user_000372|                  61| 34650.16405411844| 568.0354762970236|16.789591138629277|987.9473420832819|             14|              3|2025-03-09|
[2025-04-22T17:17:27.174+0000] {subprocess.py:93} INFO - |user_000098|                  62|30666.292624224283| 494.6176229713594|  8.20769321068607|998.0360121059581|             14|              3|2025-03-09|
[2025-04-22T17:17:27.174+0000] {subprocess.py:93} INFO - |user_000066|                  88| 45905.10992483986| 521.6489764186348| 1.651592325326979|988.6011030301456|             14|              3|2025-03-09|
[2025-04-22T17:17:27.174+0000] {subprocess.py:93} INFO - |user_000113|                  95|51318.844068761646| 540.1983586185437| 6.301721981258314|995.8016338407768|             14|              3|2025-03-09|
[2025-04-22T17:17:27.175+0000] {subprocess.py:93} INFO - +-----------+--------------------+------------------+------------------+------------------+-----------------+---------------+---------------+----------+
[2025-04-22T17:17:27.175+0000] {subprocess.py:93} INFO - only showing top 5 rows
[2025-04-22T17:17:27.175+0000] {subprocess.py:93} INFO - 
[2025-04-22T17:17:27.486+0000] {subprocess.py:93} INFO - Traceback (most recent call last):
[2025-04-22T17:17:27.487+0000] {subprocess.py:93} INFO -   File "/opt/***/external_scripts/2_calculate_features.py", line 178, in <module>
[2025-04-22T17:17:27.487+0000] {subprocess.py:93} INFO -     agg_fts.write \
[2025-04-22T17:17:27.487+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/pyspark/sql/readwriter.py", line 1398, in save
[2025-04-22T17:17:27.487+0000] {subprocess.py:93} INFO -     self._jwrite.save(path)
[2025-04-22T17:17:27.488+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/py4j-0.10.9.7-src.zip/py4j/java_gateway.py", line 1322, in __call__
[2025-04-22T17:17:27.488+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/pyspark/errors/exceptions/captured.py", line 169, in deco
[2025-04-22T17:17:27.488+0000] {subprocess.py:93} INFO -     return f(*a, **kw)
[2025-04-22T17:17:27.488+0000] {subprocess.py:93} INFO -   File "/opt/spark/python/lib/py4j-0.10.9.7-src.zip/py4j/protocol.py", line 334, in get_return_value
[2025-04-22T17:17:27.488+0000] {subprocess.py:93} INFO - py4j.protocol.Py4JError: An error occurred while calling o165.save
[2025-04-22T17:17:28.407+0000] {subprocess.py:97} INFO - Command exited with return code 1
[2025-04-22T17:17:28.991+0000] {taskinstance.py:2699} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/bash.py", line 212, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 1.
[2025-04-22T17:17:29.087+0000] {taskinstance.py:1138} INFO - Marking task as UP_FOR_RETRY. dag_id=gen_lxw_fts, task_id=calculate_fts, execution_date=20250308T230000, start_date=20250422T171425, end_date=20250422T171729
[2025-04-22T17:17:29.132+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 144 for task calculate_fts (Bash command failed. The command returned a non-zero exit code 1.; 3609)
[2025-04-22T17:17:29.166+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2025-04-22T17:17:29.201+0000] {taskinstance.py:3281} INFO - 0 downstream tasks scheduled from follow-on schedule check
